static/src/javascripts/projects/common/modules/commercial/build-page-targeting.spec.js

```diff
@@ -72,9 +76,18 @@ const tcfNullConsentMock = (callback): void =>
     callback({ '1': null, '2': null, '3': null, '4': null, '5': null });
 const tcfMixedConsentMock = (callback): void =>
     callback({ '1': false, '2': true, '3': true, '4': false, '5': true });
+
 const ccpaWithConsentMock = (callback): void => callback(false);
 const ccpaWithoutConsentMock = (callback): void => callback(true);
 
+const tcfv2WithConsentMock = (callback): void =>
+    callback({ tcfv2 : { consents: {  '1':  true , '2': true }, eventStatus: 'useractioncomplete'}});
+const tcfv2WithoutConsentMock = (callback): void =>
+    callback({ tcfv2 : { consents: { }, eventStatus: 'cmpuishown' }});
+const tcfv2MixedConsentMock = (callback): void =>
+    callback({ tcfv2 : { consents: {  '1':  true , '2': false }, eventStatus: 'useractioncomplete'}});
```

Comment:
As per the doc, the last consent will always be `true`!
```suggestion
    callback({ tcfv2 : { consents: {  '1':  true , '2': false, '3': true }, eventStatus: 'useractioncomplete'}});
```

Reply:
Oh missed that! Can you share the doc?

Reply:
From @jeteve :

> **NOTA BENE**: If we're talking about purposes, note that in case all purposes are false, the purpose object will be empty. This is explained in details [here].
> 
> In short, in case of no consent at all, the Object.values( consentObject ) will be zero sized, so it will NOT include false as a value.
>
> Examples:
>
> if your consent set is: {} , you get the consent object {}.
> if your consent set is {2}, you get the consent object: { '1': false, '2': true }. You can see this behaviour yourself by consenting to just one purpose.
> if your consent set is {2,5}, you get the consent object: {'1': false, , '2':true, '3': false, '4': false, '5': true }.

[here]: https://trello.com/c/U5hiF9vd/816-cmp-problem-gettcdata-returns-an-incorrect-consent-object#comment-5ee3fd89dc12915ef5f7027c

Reply:
Ah yeah good catch, thanks. it stops appending the falses after the last true. I have switched the order of the consents instead